device_doctor/lib/src/utils.dart

```diff
@@ -18,3 +21,33 @@ class BuildFailedError extends Error {
   @override
   String toString() => message;
 }
+
+/// Creates a directory from the given path, or multiple path parts by joining
+/// them using OS-specific file path separator.
+Directory dir(String thePath,
+    [String part2, String part3, String part4, String part5, String part6, String part7, String part8]) {
```

Comment:
Spread operator would clean this up.

https://dart.dev/guides/language/language-tour#spread-operator

Reply:
This is to align with: https://github.com/dart-lang/path/blob/master/lib/path.dart#L257